### ISSUE TITLE
Stacking Window Mode

### ISSUE THREAD

**Issue by hoebbelsB** [material-shell/material-shell#488]
First of all, thank you very much for your effort. This extension is AMAZING!

# Description

Currently there is one major missing feature which is keeping me from moving from [pop shell](https://github.com/pop-os/shell). Its the [stacking mode](https://github.com/pop-os/shell#stacking-mode).
Most of your features are far superior compared to pop-shells implementation, but the stacking mode is essential for my workflow.
I recently switched to an ultrawide screen and having one app in full-screen is simply not the way you should use it. I am a software developer and working on different projects on a daily basis. In most cases I have multiple windows of some form of IDE (VSCode, IntelliJ xy, ...) open for quick access and lookups. But in most scenarios I need them once at a time. For this reason I have all of them sitting in a "stack" on the left side of the screen. On the right there is browser, files, console, terminal and stuff. Tiled in something like a grid mode. This would be something like the `Half` display layout with a stack of applications on the left side. I tried your extension multiple times and never could achieve a similar behavior :(.

# Solution

The solution I would propose is to have an option to create Stacks of applications which tile and behave like a single window, but containing multiple windows with a distinct navigation to switch between them. One could think of a similar behavior like pop-shell does. A focused window can be set into stacking mode. Other windows now can be dragged inside of it.
Oh, of course the stack should get memorized :+1: (btw. love that feature!)

Apparently I am a decent javascript developer and am willing to contribute to your project if you wish so. But I didn't take a look at the source code yet and don't know how steep the learning curve would be. So, just let me know :)

**Comment by PapyElGringo** [comment]
Hey thanks for your feedback.

From Material Shell point of you, it's seem that you would like to have 2 tiling workspace side by side in your ultra wide monitor.
The first half of the monitor being an maximized layout tiling workspace and the second one an half tiling layout workspace.

It's somehow related to a feature I had in my mind for some time now the ability to select several workspace to display at one time.

It's could be interesting to see how we could implement that

**Comment by hoebbelsB** [comment]
Yeah you could implement it like this as well. May be even more flexible. Do you have any plans for this in the upcoming months? :)

**Comment by simongottschlag** [comment]
A workflow I can't be without on a 49" monitor is being able to stack windows. I usually have three stacks side by side, a larger one in the middle and smaller ones on the sides of it.

This makes it easy to use the middle one as a normal sized monitor and the secondary ones for things like documentation, chat etc.

It's one of the benefits with i3/sway - but they come with other limitations. When using Sway, you can only screen share the whole monitor and not specific applications which doesn't work well on 49".

**Comment by PapyElGringo** [comment]
You can use the split layout with 3 column

**Comment by simongottschlag** [comment]
@PapyElGringo and that I do! But being able to stack the different columns makes the whole endeavor much more pleasant 😊

**Comment by hakimzulkufli** [comment]
Coming from Windows, I use FancyZones ever since it was announced. Being able to choose which column we want to stack and which we don't want to stack is indeed would be more beneficial than a set of predetermined stackable and non-stackable locations. Obviously, FancyZones take it many steps further by allowing us to create a custom layout completely.

Material Shell almost had it with the "Half" layouts, but it's only full on the left side and halved on the right. If we can somehow change the full column side to be the right side, that will be the most simplistic solution for now. Or better yet, allow creation of a new column in Grid mode. I feel like Grid mode should be more flexible than only Nx2 layout. This is probably the best way to have a similar experience to the Stacking mode in popShell.

Material Shell is amazing. I love it. It's almost where I want it to be.
